### ISSUE TITLE
Superdeduper starts outputting immediately, is "-q" the default behavior?

### ISSUE THREAD

**Issue by samhunter** [ibest/HTStream#42]
Running super-deduper like:
~/HTStream/build/Super-Deduper/super_deduper -1 00-RawData/1116_S61_R1_001.fastq.gz -2 00-RawData/1116_S61_R2_001.fastq.gz -t -O

Will immediately start generating output reads even though the gzip processes are still running. This doesn't seem possible. Is super_deduper using -q mode by default and skipping the best-quality check?

**Comment by dstreett** [comment]
Yup, it is default with Q 20. Should I change that?

**Comment by samhunter** [comment]
So, the current behavior is:

Find first occurrence of a read, if average Qscore (across all q-scores in both pairs) is >= 20 keep it and skip all future reads, if averageQ is < 20 store it in memory until a read with better q-score is found?

I kind of like it, I think adding more details about the behavior to --help would be a good choice :)

**Comment by dstreett** [comment]
Exactly. If no read with a better q-score is found, it will be written at the end. Should this be the default or user input?

I think the mystery of what software does is invigorating part. :) Nah, you are spot on, once I get the N-Remover merged in and the bugs squashed, I will start writing.

**Comment by samhunter** [comment]
Okay, I'm fine with this behavior as default personally. I'm not sure how @msettles will feel. Certainly documenting it would help.

One request would be to make the "q" setting configurable. For example, perhaps I want to set q=30 as the average for a shorter-read/high quality run.

**Comment by dstreett** [comment]
Okay, it is now configurable with -q option. Default is 40, so only "perfect" reads are written out by default, however, you can set it to whatever you would like.

Since I'm being lazy, it is in the N remover commit so I will close this once it is merged into master.

**Comment by samhunter** [comment]
Cool.
